src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -265,9 +269,11 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         let preferred: KernelConnectionMetadata | undefined;
 
         if (this.isLocalLaunch) {
-            const preferredConnectionPromise = preferred
-                ? Promise.resolve(preferred)
-                : this.localKernelFinder.findKernel(document.uri, getNotebookMetadata(document), token);
+            const preferredConnectionPromise = this.localKernelFinder.findKernel(
```

Comment:
Please modify the `findKernel` method in `LocalKernelFinder.ts` to now use the `document` to determine if its an interactive window or not.

Reply:
We have code in there to return the active interpreter if resource is `*.py`, we need to change the log to inspect the notebook document instead.

Reply:
You'd want to do the same with the `remoteKernelFinder.ts`